### PR TITLE
Added backoffLimit configuration option to all scan Jobs

### LINKED ISSUE #238: Scans are marked as Failed after the First Jobs has failed
## Current Status

The secureCodeBox Operator marks Scans as "Failed" if the job fails, even when the job gets restarted by Kubernetes and succeeds later.

## Expected Behavior

The scan is only marked as "Failed" if the retry limit for the job has been exceeded.

### PR DESCRIPTION
**Pull request by rseedorff**
## Checklist

* [ ] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [ ] Make sure `npm test` runs for the whole project.

## Description

Closes #238

### CODE REVIEW

File: scanners/kubeaudit/templates/kubeaudit-scan-type.yaml [hunk 1]
```diff
@@ -8,7 +8,12 @@ spec:
     location: "/home/securecodebox/kubeaudit.jsonl"
   jobTemplate:
     spec:
-      ttlSecondsAfterFinished: 10
```
Comment (rseedorff) [inline scanners/kubeaudit/templates/kubeaudit-scan-type.yaml:11]: Is this an important default we should define?

File: scanners/git-repo-scanner/templates/git-repo-scanner-scan-type.yaml [hunk 2]
```diff
@@ -11,7 +11,9 @@ spec:
       {{- if .Values.scannerJob.ttlSecondsAfterFinished }}
       ttlSecondsAfterFinished: {{ .Values.scannerJob.ttlSecondsAfterFinished }}
       {{- end }}
-      backoffLimit: 3
+      {{- if .Values.scannerJob.backoffLimit }}
+      backoffLimit: {{ .Values.scannerJob.backoffLimit }}
+      {{- end }}
```
Comment (J12934) [inline scanners/git-repo-scanner/templates/git-repo-scanner-scan-type.yaml:16]: I'd skip the if here and directly reference the backoff limit as we have a non nil default for it.

Comment (rseedorff) [reply scanners/git-repo-scanner/templates/git-repo-scanner-scan-type.yaml]: Ok 🤷🏻‍♂️

File: scanners/kubeaudit/templates/kubeaudit-scan-type.yaml (hunk 1 above)
Comment (J12934) [reply scanners/kubeaudit/templates/kubeaudit-scan-type.yaml:11]: Mh no this seems to have been forgotten by the ttl pull request.
Good catch

**Review by J12934 — APPROVED** [review]

**Pull request merged**